epictrack-web/src/components/project/ProjectList.tsx

```diff
@@ -50,6 +52,21 @@ const ProjectList = () => {
 
   const projects = React.useMemo(() => ctx.data as Project[], [ctx.data]);
 
+  async function fetchProponents() {
+    const project_proponents = projects
+      .map((p) => p.proponent_id)
+      .filter((ele, index, arr) => arr.findIndex((t) => t === ele) === index);
+
+    const proponentsResponses = await Promise.all(
+      project_proponents.map((proponentId) =>
+        proponentService.getById(proponentId.toString())

```

Comment:
can we try to find another solution for this other than doing multiple API Calls, one for each proponent Id, can we fetch the whole list from the backend?

Reply:
yea ill do that then ill create a backend route to get all the ids